message.go

```diff
@@ -19,6 +21,9 @@ type Message struct {
 	// If not set at the creation, Time will be automatically set when
 	// writing the message.
 	Time time.Time
+
+	// Compression codec used to encode the message value
+	CompressionCodec int8
```

Comment:
What's your take on using a `CompressionCodec` here instead of the code?

Reply:
You mean the type CompressionCodec ?

Reply:
Yep, basically carrying an instance of `CompressionCodec` instead of the code.

Reply:
With the actual implementation it means that we need a type of the "None" codec case.

Reply:
Can we use nil to mean there is no compression?